src/renderer/features/discord-rpc/use-discord-rpc.ts

```diff
@@ -136,8 +136,17 @@ export const useDiscordRpc = () => {
     useEffect(() => {
         if (!discordSettings.enabled) return discordRpc?.quit();
 
-        discordRpc?.initialize(discordSettings.clientId);
+        let timeout: any;
+        (async function initialize() {
+            // Checks RPC status every 10 seconds, initializes if needed
+            const isConnected = await discordRpc?.isConnected();
+            if (!isConnected) {
+                discordRpc?.initialize(discordSettings.clientId);
+            }
+            timeout = setTimeout(initialize, 10000);
+        })();
         return () => {
+            clearTimeout(timeout);
             discordRpc?.quit();
```

Comment:
I'm personally not a fan of the interval approach.

I would rather have it so that it's checked at the top of `setActivity` so that it validates and initializes before every call.